### ISSUE TITLE
Some users get trapped in the "fullscreen" / "hidden UI" mode

### ISSUE THREAD

**Issue by pastk** [organicmaps/organicmaps#9344]
A long tap on the map makes all UI buttons disappear. To bring the UI back a user has to long tap again.

Several users reported they entered the fullscreen (FS) mode inadvertently and had to restart OM to get out of it.
https://fosstodon.org/@organicmaps/113163548649933395
(@rtsisyk said there were a few support emails also)

There is a toast shown on the first feature use `Long-tap on the map again to see the interface`. Obviously its not enough.

**Comment by pastk** [comment]
Some ideas for improvement:
- make the toast pop up on every use (IMHO it won't be enough still)
- switch the FS mode by a small corner button
 ![image](https://github.com/organicmaps/organicmaps/assets/18434508/dc0aa045-03d9-4a4c-9248-359432b7060c)
- make long tap action customizable (was discussed elsewhere) and the default action shouldn't be "hide the ui"
- use **any tap** to leave the FS mode https://github.com/organicmaps/organicmaps/pull/8551#issuecomment-2188295542

**Comment by pastk** [comment]
Context: the "hide UI" feature is quite niche but is useful in some cases
![image](https://github.com/user-attachments/assets/c8bb2dbb-b43c-4243-a629-3e861941bce1)
https://t.me/organicmaps/41938

**Comment by pastk** [comment]
To my mind the best solution would be to use a corner button to exit the FS mode.
To enter FS the same button could be used (might clutter the UI for users who don't need this feature) or maybe better to add a separate "Hide UI" button into the Layers dialog.

Customizing long tap action is a more generic feature and could be complimentary to other solutions.

**Comment by hoizan** [comment]
Honestly, I find that the current implementation is simple and easy to use. To me, it is intuitive enough to discover. Unless there are more compelling reasons or use cases, I will be upset to see that this full screen feature by long tapping being replaced.

**Comment by hoizan** [comment]
I believe bu slightly increase the tap duration, say by 0.5sec, will  solve this issue.

**Comment by auge8472** [comment]
> Honestly, I find that the current implementation is simple and easy to use. To me, it is intuitive enough to discover.

That is a weak argument. The fact that *you* understand how it works is great for *you*, but not for others. How are you supposed to know that the long tap exists and what it does? The fact that it was displayed as a tooltip at some point in the past is honestly no help (and I'm sure, I've not seen it). How is someone supposed to know how to get out of it if one doesn't even know how one fell into this trap? This more or less hidden feature is anything but self-explanatory.

> Unless there are more compelling reasons or use cases, I will be upset to see that this full screen feature by long tapping being replaced.

IMHO the poor comprehensibility is such a "compelling reason". And tapping a half second longer to activate it makes it not better.

**Comment by hoizan** [comment]
> > Honestly, I find that the current implementation is simple and easy to use. To me, it is intuitive enough to discover.
> 
> That is a weak argument. The fact that _you_ understand how it works is great for _you_, but not for others. How are you supposed to know that the long tap exists and what it does? The fact that it was displayed as a tooltip at some point in the past is honestly no help (and I'm sure, I've not seen it). How is someone supposed to know how to get out of it if one doesn't even know how one fell into this trap? This more or less hidden feature is anything but self-explanatory.
> 
> > Unless there are more compelling reasons or use cases, I will be upset to see that this full screen feature by long tapping being replaced.
> 
> IMHO the poor comprehensibility is such a "compelling reason". And tapping a half second longer to activate it makes it not better.


I am not arguing on anything. I am saying that the implementation is easy enough to be discovered by most users, and I prefer it. Normal tapping and long tapping are very basic UI interfaces for touch screen operations. For instance, a long tap on many apps, may offer a pop-up a menu for users to select options. For OM, the long tap now hides the menu buttons to present a full uncluttered screen, which is not obscure to comprehend at all.

Everyone is entitled to express one's opinions and preferences. Ultimately, the developers are the ones to decide which are the best implementations to serve most users.

**Comment by auge8472** [comment]
> Everyone is entitled to express one's opinions and preferences.

That's what I did. ;-)

**Comment by map-per** [comment]
> - use **any tap** to leave the FS mode https://github.com/organicmaps/organicmaps/pull/8551#issuecomment-2188295542

I think that's a good solution. I just checked an old OM version. Before single and long tap were swapped selecting a place used to exit the full screen mode. So this approach would restore the old behavior.

**Comment by pastk** [comment]
> or maybe better to add a separate "Hide UI" button into the Layers dialog.

...or maybe hide the UI upon a long tap on a layers button. But it won't be discoverable.

**Comment by map-per** [comment]
Today a friend of mine complained that the "buttons sometimes disappear when using OM for too long without closing the app". They did not know how to leave the full-screen mode so they used to close the app every time this happened.

**Comment by eisams** [comment]
I did a check on Apple Maps and Google Maps, and they don't have this functionality for hiding the UI

If you long-tap in e.g,. Apple Maps you drop a pin on the map if it's not a highlighted POI you press on
In Google Maps you also just select whatever object its on (and there's no "long tap")

So this could explain why people get stuck, it's a completely new UX feature

**Comment by cpmsmith** [comment]
> In Google Maps you also just select whatever object its on (and there's no "long tap")

I believe that Google Maps used to have the reverse of what OM has now – short tap on a "blank" area to go full screen, and long tap on a blank area to drop a pin – and [they removed it](https://www.phonearena.com/news/google-maps-full-screen-map_id149751). Now, you have to swipe the search bar at the top of the screen upward. 

Crucially, in Google Maps, if you do tap a POI and bring up the details for it, it exits full screen, and in my opinion OM should do the same. It would solve the problem where people just tap everything onscreen to no avail, and once you have the details for a POI open, being in full screen makes very little difference anyway:

<img src="https://github.com/user-attachments/assets/acf72fbb-abe3-4c60-b4dc-a736e1f4273b" width="375">
<img src="https://github.com/user-attachments/assets/86ac496e-1894-4c46-b929-7155d7d40263" width="375">

**Comment by pastk** [comment]
> Today a friend of mine complained that the "buttons sometimes disappear when using OM for too long without closing the app". They did not know how to leave the full-screen mode so they used to close the app every time this happened.

Here is an easy incremental improvement until a more elaborate solution is implemented:
- https://github.com/organicmaps/organicmaps/pull/9902

**Comment by dvdmrtnz** [comment]
> Crucially, in Google Maps, if you do tap a POI and bring up the details for it, it exits full screen, and in my opinion OM should do the same. It would solve the problem where people just tap everything onscreen to no avail, and once you have the details for a POI open, being in full screen makes very little difference anyway:

I agree 100% with this. In my opinion, tapping on a POI to bring up the details is most intuitive way to exit from fullscreen mode.
